File: android/src/main/java/cn/stormyang/system_boot_time/SystemBootTimePlugin.java
package cn.stormyang.system_boot_time;

import android.os.SystemClock;

import androidx.annotation.NonNull;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import cn.stormyang.system_boot_time.SystemBootTimeSpace.SystemBootTime;

/** SystemBootTimePlugin */
public class SystemBootTimePlugin implements FlutterPlugin, SystemBootTimeSpace.SystemBootTime {

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
    SystemBootTime.setup(binding.getBinaryMessenger(), this);
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    SystemBootTime.setup(binding.getBinaryMessenger(), null);
  }

  @Override
  public Long second() {
    return SystemClock.elapsedRealtime() / 1000;
  }
}

File: android/src/main/kotlin/cn/stormyang/flutter/system_boot_time/system_boot_time/SystemBootTimePlugin.kt
package cn.stormyang.flutter.system_boot_time.system_boot_time

import android.os.Build
import android.os.SystemClock
import androidx.annotation.NonNull
import androidx.annotation.RequiresApi

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** SystemBootTimePlugin */
class SystemBootTimePlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "system_boot_time")
    channel.setMethodCallHandler(this)
  }

  @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "getSystemBootTime") {
      result.success(SystemClock.elapsedRealtime() / 1000)
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
